src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IFederationWalletManager.cs

```diff
@@ -142,7 +134,10 @@ public interface IFederationWalletManager
         /// Enables federation.
         /// </summary>
         /// <param name="password">The federation wallet password.</param>
-        void EnableFederation(string password);
+        /// <param name="mnemonic">The user's mnemonic.</param>
+        /// <param name="passphrase">A passphrase used to derive the private key from the mnemonic.</param>
+        /// <param name="onlyImportKey">If set to <c>true</c> imports the key without activating the federation.</param>
+        void EnableFederation(string password, string mnemonic = null, string passphrase = null, bool onlyImportKey = false);
```

Comment:
I think you can remove this `bool onlyImportKey = false`. Why wouldn't we enable the federation?

Reply:
to remain compatible with the fact that we were not doing it before during the import key.